src/annotator/integrations/test/image-text-layer-test.js

```diff
@@ -0,0 +1,236 @@
+      const charBoxes = createCharBoxes(imageText).slice(0, -1);
+      createTextLayer(image, charBoxes, imageText);
+    }, 'Char boxes length does not match text length');
+  });
+
+  it('creates elements in the text layer for each word in the image', () => {
+    const { image } = createPageImage();
+    const imageText = 'some words here';
+    const textLayer = createTextLayer(
+      image,
+      createCharBoxes(imageText),
+      imageText
+    );
+
+    assert.equal(textLayer.container.textContent, 'some words here ');
+    const wordSpans = getWordBoxes(textLayer);
+    assert.equal(wordSpans.length, imageText.split(' ').length);
+    assert.deepEqual(
+      wordSpans.map(ws => ws.textContent),
+      ['some', 'words', 'here']
+    );
+
+    const imageBox = image.getBoundingClientRect();
+    const wordBoxPositions = wordSpans.map(span => {
+      const wordBox = span.getBoundingClientRect();
+      return [
+        wordBox.left - imageBox.left,
+        wordBox.top - imageBox.top,
+        wordBox.width,
+        wordBox.height,
+      ].map(coord => Math.floor(coord));
+    });
+
+    const imageWidth = parseInt(image.style.width);
+    const imageHeight = parseInt(image.style.height);
+
+    const expectedPositions = [
+      expectedBoxOffsetAndSize(imageWidth, imageHeight, 0, 'some'),
+      expectedBoxOffsetAndSize(imageWidth, imageHeight, 1, 'words'),
+      expectedBoxOffsetAndSize(imageWidth, imageHeight, 2, 'here'),
+    ];
+    assert.deepEqual(wordBoxPositions, expectedPositions);
+  });
+
+  it('updates size and position of text layer elements when window is resized', () => {
+    const { image } = createPageImage();
+    const imageText = 'some text in the image';
+
+    const clock = sinon.useFakeTimers();
+    try {
+      const textLayer = createTextLayer(
+        image,
+        createCharBoxes(imageText),
+        imageText
+      );
+
+      const originalBoxes = getWordBoxes(textLayer).map(box =>
+        box.getBoundingClientRect()
+      );
+
+      // Rescale image to 3/5 of original size.
+      image.style.width = '300px';
+      image.style.height = '300px';
+
+      // Notify text layer that image has been resized. We currently assume
+      // that this always corresponds to a window resize.
+      window.dispatchEvent(new Event('resize'));
+      clock.tick(100);
+
+      // Check that the positions and sizes of each text box were changed to
+      // reflect the new scale of the image.
+      const ratio = 3 / 5;
+      const imageBox = image.getBoundingClientRect();
+      const newBoxes = getWordBoxes(textLayer).map(box =>
+        box.getBoundingClientRect()
+      );
+
+      const tolerance = 0.01;
+      assert.equal(originalBoxes.length, newBoxes.length);
+      for (let [i, originalBox] of originalBoxes.entries()) {
+        const newBox = newBoxes[i];
+
+        const leftGap = originalBox.left - imageBox.left;
+        const newLeftGap = newBox.left - imageBox.left;
```

Comment:
> In my opinion this is the same: assert.approximately(originalBox.left * ratio, newBox.left, tolerance)

This isn't correct as the `ratio` scaling applies only to the gap between the edge of the image and the text box, not the larger gap between the edge of the viewport and the text box. The test does currently pass using this expression, but that's because the image's left and top edges are at 0 and 0 in the viewport.